lib/rubocop/cop/style/mutable_constant.rb

```diff
@@ -70,8 +121,49 @@ def mutable_literal?(value)
           value && value.mutable_literal?
         end
 
+        def immutable_literal?(node)
+          node.nil? || node.immutable_literal?
+        end
+
+        def frozen_string_literal?(node)
+          FROZEN_STRING_LITERAL_TYPES.include?(node.type) &&
+            frozen_string_literals_enabled?
+        end
+
+        def requires_parentheses?(node)
+          node.irange_type? ||
```

Comment:
I think there is a good case for adding `Node#range_type?`, but that could be considered out of scope for this change. 🙂